sonaranalyzer-dotnet/tests/SonarAnalyzer.UnitTest/TestCases/VariableShadowsField.cs

```diff
@@ -76,4 +76,37 @@ public static VariableShadowsField build(int MyField)
             return null;
         }
     }
+
+    public class WithLocalFunctions
+    {
+        private int field;
+
+        public void Method()
+        {
+            void doSomething()
+            {
+                int field = 0; // Noncompliant
+            }
+
+            static void doMore()
+            {
+                int field = 0; // Noncompliant
+            }
+        }
+
+        public void MethodWithLocalVar()
+        {
+            bool isUsed = true;
+
+            void doSomething()
+            {
+                bool isUsed = true; // Compliant - FN: should check also function variable shadowing?
+            }
+
+            static void doMore()
+            {
+                bool isUsed = true; // Compliant - FN: should check also function variable shadowing?
```

Comment:
if it's static, it won't capture the context, so in this case it should be ok, I believe... right?

Reply:
Indeed the context is not captured but it can still be misleading. Part of the rule specifies: 

> shadowing a variable declared in an outer scope can strongly impact the readability, and therefore the maintainability, of a piece of code.

From another point of view since the context is not shared don't know if this really qualifies as shadowing :)

I will remove the comment (leave only `// Compliant`) and we will decide how to proceed when we fix the other case.

Reply:
I guess shadowing refers to hiding what could otherwise be accessible.

Indeed it would be misleading, but the message of the rule woulnd't really apply, as it wouldn't truly be shadowing